components/esp_wifi_remote/eppp/wifi_remote_rpc_client.cpp

```diff
@@ -0,0 +1,277 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <netdb.h>
+#include <memory>
+#include <cinttypes>
+#include "esp_log.h"
+#include "esp_tls.h"
+#include "esp_wifi.h"
+#include "esp_check.h"
+#include "wifi_remote_rpc_impl.hpp"
+#include "eppp_link.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/event_groups.h"
+#include "wifi_remote_rpc_params.h"
+
+extern "C" esp_netif_t *wifi_remote_eppp_init(eppp_type_t role);
+
+namespace eppp_rpc {
+
+namespace client {
+const char *TAG = "rpc_client";
+
+const unsigned char ca_crt[] = "-----BEGIN CERTIFICATE-----\n" CONFIG_ESP_WIFI_REMOTE_EPPP_SERVER_CA "\n-----END CERTIFICATE-----";
+const unsigned char crt[] = "-----BEGIN CERTIFICATE-----\n" CONFIG_ESP_WIFI_REMOTE_EPPP_CLIENT_CRT "\n-----END CERTIFICATE-----";
+const unsigned char key[] = "-----BEGIN RSA PRIVATE KEY-----\n" CONFIG_ESP_WIFI_REMOTE_EPPP_CLIENT_KEY "\n-----END RSA PRIVATE KEY-----";
+
+}
+
+using namespace client;
+
+struct Sync {
+    void lock()
+    {
+        xSemaphoreTake(mutex, portMAX_DELAY);
+    }
+    void unlock()
+    {
+        xSemaphoreGive(mutex);
+    }
+    esp_err_t init()
+    {
+        mutex = xSemaphoreCreateMutex();
+        events = xEventGroupCreate();
+        return mutex == nullptr || events == nullptr ? ESP_ERR_NO_MEM : ESP_OK;
+    }
+    esp_err_t wait_for(EventBits_t bits, uint32_t timeout = portMAX_DELAY)
+    {
+        return xEventGroupWaitBits(events, bits, pdTRUE, pdTRUE, timeout) == bits ? ESP_OK : ESP_FAIL;
+    }
+    esp_err_t notify(EventBits_t bits)
+    {
+        xEventGroupSetBits(events, bits);
+        return ESP_OK;
+    }
+    ~Sync()
+    {
+        if (mutex) {
+            vSemaphoreDelete(mutex);
+        }
+        if (events) {
+            vEventGroupDelete(events);
+        }
+    }
+    SemaphoreHandle_t mutex{nullptr};
+    EventGroupHandle_t events{nullptr};
+    const int request = 1;
+    const int resp_header = 2;
+    const int resp_payload = 4;
```

Comment:
Will this work any better?
```suggestion
  enum class Access {
    request = 1,
    resp_header = 2,
    resp_payload = 4
  };
```

Reply:
ok, will do.

Reply:
I left the const variables here, as I need them as integers for the event group.